src/content/zero-knowledge-proofs.md

```diff
@@ -143,6 +143,39 @@ This is where verifiable computation comes into play. When a node executes a tra
+
+For example, imagine there are two options up for voting on a ballot (`option A` and `option B`). Alice prefers `option A`, but votes for `option B` after receiving a bribe from Bob. Here, Alice and Bob have colluded to mislead others about the popularity of `option A` to the detriment of `option B`. 
+
+Bribery and collusion limit the effectiveness of certain use-cases enabled by blockchain voting, especially [quadratic funding](https://www.radicalxchange.org/concepts/plural-funding/). Fortunately, newer solutions, such as MACI (Minimum Anti-Collusion Infrastructure), are using zero-knowledge proofs to mitigate this problem and increase the integrity of on-chain voting. 
+
+At its core, MACI is a system of smart contracts and scripts that allow a central administrator (called a "coordinator") to aggregate votes and tally results *without* revealing specifics on how each individual voted. Even so, it is still possible to verify that the votes were counted properly, or confirm that a particular individual participated in the voting round. 
+
+#### How does MACI work with zero-knowledge proofs? {#how-maci-works-with-zk-proofs}
+
+At the start, the coordinator deploys the MACI contract on Ethereum, after which users can sign up for voting (by registering their public key in the smart contract). Users cast votes by sending messages encrypted with their public key to the smart contract (a valid vote must be signed with the most recent public key associated with the user's identity, among other criteria). Afterward, the coordinator processes all messages once the voting period ends, tallies the votes, and verifies the results on-chain. 
+
+In MACI, zero-knowledge proofs are used to ensure correctness of computation by making it impossible for the coordinator to incorrectly process votes and tally results. This is achieved by requiring the coordinator to generate ZK-SNARK proofs verifying that a) all messages were processed correctly b) the final result corresponds to the sum of all *valid* votes. 
+
+Thus, even without sharing a breakdown of votes per user (as is usually the case), MACI guarantees the integrity of results calculated during the tallying process. This feature is useful in reducing the effectiveness of basic collusion schemes. We can explore this possibility by using the previous example of Bob bribing Alice to vote for an option: 
+
+- Alice agrees to vote for `option B` in exchange for a bribe from Bob
+- Alice secretly sends a transaction to change the public key associated with her identity. 
```

Comment:
Maybe mention _how_ this is secret given that blockchain is normally transparent. The message is just encrypted, correct?

Reply:
Just added the bit about encryption.